cpp/src/aztec/honk/sumcheck/sumcheck.hpp

```diff
@@ -60,16 +61,35 @@ template <class Multivariates, class Transcript, template <class> class... Relat
             round_univariate = round.compute_univariate(multivariates.folded_polynomials, alpha);
             transcript.add_element("univariate_" + std::to_string(multivariates.multivariate_d - round_idx),
                                    round_univariate.to_buffer());
-
-            transcript.apply_fiat_shamir("u_" + std::to_string(multivariates.multivariate_d - round_idx));
-            multivariates.fold(
-                multivariates.folded_polynomials, multivariates.multivariate_n, round_challenges[round_idx]);
+            challenge_label = "u_" + std::to_string(multivariates.multivariate_d - round_idx);
+            transcript.apply_fiat_shamir(challenge_label);
+            FF round_challenge = FF::serialize_from_buffer(transcript.get_challenge(challenge_label).begin());
+            multivariates.fold(multivariates.folded_polynomials, round.round_size, round_challenge);
+            round.round_size = round.round_size >> 1;
         }
 
         // Final round
-        // TODO(Cody): Figure out how to actually handle this. Currently done in prover, should happen here prob.
-        // auto multivariate_evaluations = multivariates.batch_evaluate();
-        // transcript.add_element("multivariate_evaluations", to_buffer(multivariate_evaluations));
+        transcript.add_element("multivariate_evaluations",
```

Comment:
This was done before in prover.cpp, adding each multivariate on its own by looping over the polynomial manifest. That approach required also looping over a polynomial manifest in the sumcheck verifier. To avoid providing a manifest to the sumcheck module, I opted to take this approach.